pkg/extensions/cluster_test.go

```diff
@@ -62,15 +72,288 @@ var _ = Describe("Cluster", func() {
 		})
 	})
 
-	Describe("#GetShootStateForCluster", func() {
-		var (
-			expectedShoot      *gardencorev1beta1.Shoot
-			expectedShootState *gardencorev1beta1.ShootState
+	Describe("#SyncClusterResourceToSeed", func() {
+		BeforeEach(func() {
+			expectedCloudProfile = &gardencorev1beta1.CloudProfile{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "CloudProfile",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			expectedSeed = &gardencorev1beta1.Seed{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Seed",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			expectedShoot = &gardencorev1beta1.Shoot{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Shoot",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "garden-bar",
+				},
+			}
+
+			clusterName = "shoot--" + expectedShoot.Namespace + "--" + expectedShoot.Name
+			cluster = &extensionsv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterName,
+				},
+			}
+		})
+
+		It("should not sync cloudprofile, shoot and seed to cluster if seed is not assigned to shoot", func() {
+			Expect(SyncClusterResourceToSeed(ctx, fakeSeedClient, cluster.Name, expectedShoot, expectedCloudProfile, expectedSeed)).To(Succeed())
+			Expect(fakeSeedClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster)).To(BeNotFoundError())
 
-			clusterName string
-			cluster     *extensionsv1alpha1.Cluster
-		)
+			Expect(cluster.Spec.CloudProfile.Object).To(BeNil())
```

Comment:
This and the lines after that are no-longer necessary because the cluster is not retrieved via the client now (it doesn't exist), right?

Reply:
It can but it is like a secondary check.

Reply:
Hmm, but this way we are essentially just verifying the cluster that we set in `BeforeEach` 😅

Reply:
Adapted in 6922a87cf13b19a9f795617a0f867762dd525153